### DISCUSSION TITLE
Is Aniseed's compilation incremental?

### DISCUSSION THREAD

**Discussion by alexmozaidze** [Olical/aniseed#146] [Q&A] [answered]
I'm curious to wether it blindly compiles every fennel file or does it track which files changed and only compiles them?

**Comment by Olical** [comment, marked as answer]
It compiles all files to keep things simpler, [nfnl](https://github.com/Olical/nfnl) on the other hand only compiles the file you just edited unless it's a macro file, in which case it compiles everything to make sure the macro's effects propagate. I do not attempt to parse Fennel / Lua to work out the dependencies, it's not really worth it when compile times of big projects like Conjure are measured in a handful of milliseconds, the Fennel compiler is REALLY fast because it's a pretty simple transformation in many ways.
